src/cdk-lib/opensearchserverless/vector-collection.ts

```diff
@@ -41,39 +83,250 @@ export interface VectorCollectionProps {
+  public static fromCollectionName(constructScope: Construct, constructId: string, collectionName: string): IVectorCollection {
+    class Import extends VectorCollectionBase {
+      public readonly collectionArn: string;
+      public readonly collectionId: string;
+      public readonly collectionName: string;
+      public readonly standbyReplicas: VectorCollectionStandbyReplicas;
+      public readonly aossPolicy: iam.ManagedPolicy;
+      public readonly dataAccessPolicy: oss.CfnAccessPolicy;
+
+      constructor(scope: Construct, id: string) {
+        super(scope, id);
+
+        this.collectionId = collectionName;
+        this.collectionName = collectionName;
+        this.collectionArn = cdk.Stack.of(scope).formatArn({
+          service: 'aoss',
+          resource: 'collection',
+          resourceName: collectionName,
+        });
+        this.standbyReplicas = VectorCollectionStandbyReplicas.ENABLED;
+
+        this.aossPolicy = new iam.ManagedPolicy(this, 'ImportedAOSSPolicy', {
+          statements: [
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: ['aoss:APIAccessAll'],
+              resources: [this.collectionArn],
+            }),
+          ],
+        });
+
+        this.dataAccessPolicy = new oss.CfnAccessPolicy(this, 'ImportedDataAccessPolicy', {
+          name: generatePhysicalNameV2(this, 'DataAccessPolicy', { maxLength: 32, lower: true }),
+          type: 'data',
+          policy: '[]',
+        });
+      }
+    }
+    return new Import(constructScope, constructId);
+  }
+
+  /**
+   * Import an existing collection using its attributes.
+   * @param constructScope The parent creating construct.
+   * @param constructId The construct's name.
+   * @param attrs The collection attributes to use.
+   */
+  public static fromCollectionAttributes(
+    constructScope: Construct,
+    constructId: string,
+    attrs: VectorCollectionAttributes,
+  ): IVectorCollection {
+    class Import extends VectorCollectionBase {
+      public readonly collectionArn = attrs.collectionArn;
+      public readonly collectionId = attrs.collectionId;
+      public readonly collectionName = attrs.collectionName;
+      public readonly standbyReplicas = attrs.standbyReplicas;
+      public readonly aossPolicy: iam.ManagedPolicy;
+      public readonly dataAccessPolicy: oss.CfnAccessPolicy;
+
+      constructor(scope: Construct, id: string) {
+        super(scope, id);
+
+        this.aossPolicy = new iam.ManagedPolicy(this, 'ImportedAOSSPolicy', {
+          statements: [
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: ['aoss:APIAccessAll'],
+              resources: [this.collectionArn],
+            }),
+          ],
+        });
+
+        this.dataAccessPolicy = new oss.CfnAccessPolicy(this, 'ImportedDataAccessPolicy', {
+          name: generatePhysicalNameV2(this, 'DataAccessPolicy', { maxLength: 32, lower: true }),
```

Comment:
The calls to ```generatePhysicalNameV2``` needs to be updated in this file as the function doesn't produce unique names. A new parameter was introduce and is used for instance here: https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/src/cdk-lib/bedrock/data-sources/s3-data-source.ts#L107 . As a test, try to provision 2 different collections in the same stack, with the current code it will fail

Reply:
i couldn't completely get the strong-type of `destroyCreate` field. as i understood hash value of field is being used. however, which props of VectorCollection i am supposed to use?